### PR TITLE
Improve plugins typings (Part 2)

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

See https://github.com/babel/babel/pull/14601 for more info. This PR can be reviewed by commits: each of them deal with a single plugin.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14639"></a>

### CODE REVIEW

File: packages/babel-plugin-transform-block-scoping/src/index.ts [hunk 1]
```diff
@@ -109,21 +109,27 @@ export default declare((api, opts: Options) => {
...
+const buildRetCheck = template.statement(`
   if (typeof RETURN === "object") return RETURN.v;
 `);
 
-function isBlockScoped(node) {
+function isBlockScoped(node: t.Node): node is t.VariableDeclaration {
   if (!t.isVariableDeclaration(node)) return false;
-  if (node[t.BLOCK_SCOPED_SYMBOL]) return true;
+  if (
+    // @ts-expect-error Fixme: avoid mutating AST nodes
```
Comment (JLHwung) [inline packages/babel-plugin-transform-block-scoping/src/index.ts:127]:
```suggestion
    // @ts-expect-error Fixme: document symbol properties
```

**Review by nicolo-ribaudo — APPROVED** [review]

File: packages/babel-helper-builder-binary-assignment-operator-visitor/src/index.ts [hunk 2]
```diff
@@ -3,7 +3,13 @@ import { assignmentExpression, sequenceExpression } from "@babel/types";
 import type { Visitor } from "@babel/traverse";
 import type * as t from "@babel/types";
 
-export default function (opts: { build: Function; operator: string }) {
+export default function (opts: {
+  build: (
+    left: t.Expression | t.PrivateName,
+    right: t.BinaryExpression["right"],
```
[the remark this answers is not in the document]
Comment (JLHwung) [reply packages/babel-helper-builder-binary-assignment-operator-visitor/src/index.ts:9]: 👍 Yes it is. I tried lowering `left` to `t.Expression` when operator is not `in` but failed.

File: packages/babel-plugin-transform-computed-properties/src/index.ts [hunk 3]
```diff
@@ -34,28 +45,31 @@ export default declare((api, options: Options) => {
...
+      t.expressionStatement(
+        t.assignmentExpression(
+          "=",
+          t.memberExpression(
+            t.cloneNode(objId),
+            prop.key,
+            prop.computed || t.isLiteral(prop.key),
           ),
+          // @ts-expect-error todo(flow->ts): double-check type error
+          getValue(prop),
```
[the remark this answers is not in the document]
Comment (JLHwung) [reply packages/babel-plugin-transform-computed-properties/src/index.ts:63]: `getValue` returns the `value` of `ObjectProperty`, which contains Pattern elements. It should be fixed in `getValue` then.

**Review by liuxingbaoyu — APPROVED** [review]
This looks fine to me, but since I'm not familiar with plugins it's likely that I'm overlooking something.

**Pull request merged**